js/main.js

```diff
@@ -19,13 +27,137 @@ function getGhibliMovies() {
+  $textArea.setAttribute('type', 'text-area');
+  $textArea.setAttribute('id', 'review');
+  $textArea.setAttribute('rows', '35');
+  $textArea.setAttribute('name', 'review');
+  $textArea.setAttribute('placeholder', 'Write your review here!');
+  $textArea.setAttribute('class', 'review-box');
+  $textArea.required = true;
+
+  $reviewForm.appendChild($textArea);
+  $reviewForm.appendChild($saveButton);
+
+  $columnDiv2.appendChild($reviewTitle);
+  $columnDiv2.appendChild($reviewForm);
+
+  $columnDiv.appendChild($title);
+  $columnDiv.appendChild($descriptionTitle);
+
+  $rowDiv.appendChild($columnDiv);
+  $rowDiv.appendChild($columnDiv2);
+
+  $reviewForm.addEventListener('submit', saveReview);
+
+  var dataObject = {
+    title: $search.elements.title.value,
+    image: '',
+    description: '',
+    review: ''
+  };
+
+  function saveReview(event) {
+    event.preventDefault();
+    dataObject.entryId = ghibliData.nextReviewId;
+    ghibliData.nextReviewId++;
+    dataObject.review = $reviewForm.elements.review.value;
+    ghibliData.reviews.unshift(dataObject);
+    $reviewParent.removeChild($rowDiv);
+    switchView('home-screen');
+  }
+
+  function getPoster(movieName) {
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', 'https://api.themoviedb.org/3/search/movie?api_key=e2317d1150ebe694b173d9560f5e95b8&query=' + movieName);
+    xhr.responseType = 'json';
+    xhr.addEventListener('load', function () {
+      var $posterLink = document.createElement('img');
+      var posterPath;
+      for (var i = 0; i < xhr.response.results.length; i++) {
+        if (xhr.response.results[i].title === 'Spirited Away' || xhr.response.results[i].title === 'Grave of the Fireflies' || xhr.response.results[i].title === 'Pom Poko') {
+          posterPath = xhr.response.results[1].poster_path;
+        } else {
+          posterPath = xhr.response.results[0].poster_path;
+        }
+      }
+      $posterLink.setAttribute('src', 'https://image.tmdb.org/t/p/w500' + posterPath);
+      $posterLink.setAttribute('class', 'ghibli-image');
+      $columnDiv.insertBefore($posterLink, $descriptionTitle);
+      dataObject.image = $posterLink.getAttribute('src');
+    });
+    xhr.send();
+  }
+  function getDescription() {
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', 'https://ghibliapi.herokuapp.com/films');
+    xhr.responseType = 'json';
+    xhr.addEventListener('load', function () {
+      for (var i = 0; i < xhr.response.length; i++) {
+        if ($title.textContent === xhr.response[i].title) {
+          var description = xhr.response[i].description;
+          var $description = document.createElement('div');
+          $description.setAttribute('class', 'description-text');
+          $description.textContent = description;
+          $columnDiv.appendChild($description);
+          dataObject.description = $description.textContent;
+        }
+      }
+    });
+    xhr.send();
+  }
```

Comment:
Why are these function definitions nested in `manipulateForm`? You should keep all function definitions in the global scope for now. There are some reasons to do this, but it shouldn't be necessary here. If you are doing this to have access to some specific piece of data, you should adapt these functions to take in a parameter, so you can pass that specific data as an argument.